### ISSUE TITLE
Detailslist: After checking "Toggle selection for all users" checkbox in the grid, while navigating through each row, Narrator is not intimating to screen reader users that checked users got unchecked.

### ISSUE THREAD

**Issue by lezhang1** [microsoft/fluentui#17578]
### Environment Information

- **Package version(s)**: 7.154.0
- **Browser and OS versions**: Edge latest on Win 10

### Describe the issue:

For detail list, use tab to select all rows, when navigate away(using key down), all the checked rows will get unchecked, but the screen reader is not announcing that the checked rows got unchecked or not

#### Actual behavior:

After checking "Toggle selection for all rows" checkbox in the grid, while navigating through each row, Narrator/NVDA is not intimating to screen reader users that checked users got unchecked.

#### Expected behavior:

Narrator/NVDA should announce that the rows which are not focused got unchecked when navigated to any row in the grid.

### Documentation describing expected behavior

Narrator/NVDA should announce that the rows which are not focused got unchecked when navigated to any row in the grid.

**Comment by gouttierre** [comment]
@lezhang1 - Thanks for reaching out to us and providing context into your issue.

@TristanWatanabe - Would you be able to confirm if this is a regression, or if this behavior is an issue with the unchecked/checked rows?
cc @smhigley
